benchmark/EstimateMaxCount.t.sol

```diff
@@ -65,7 +65,7 @@ contract EstimateMaxCount is Test {
 
     /// @notice Estimate the maximum number of tranches allowed in LockupTranched.
     function test_EstimateTranches() public {
```

Comment:
I'd preferred to keep it because right now, even though the segments count is less than tranches count but there could be a case in the future where tranches count may exceed segments count. So instead of rewriting it again, I'd leave the implementation.

A better approach would be to take the minimum of tranches count and segments count and update it, which is also something I intent to do (added to the OP).

Reply:
> could be a case in the future where tranches count may exceed segments count

how exactly? a tranche has the same variables as the segments expect for the exponent


> A better approach would be to take the minimum of tranches count and segments count and update it

thought about it but i find it unneeded

Reply:
> a tranche has the same variables as the segments expect for the exponent

Yes but the computation part can differ. If you are confident that segments will always be more gas expensive then we can remove it. I am doubtful so I thought the minimum of the two would be a safer choice. I leave it for you to decide.